parametric/test_128_bit_traceids.py

```diff
@@ -0,0 +1,329 @@
+import pytest
+
+from parametric.utils.headers import make_single_request_and_get_inject_headers
+from parametric.utils.test_agent import get_span
+
+parametrize = pytest.mark.parametrize
+
+
+@pytest.mark.parametrize(
+    "library_env", [{"DD_TRACE_PROPAGATION_STYLE": "Datadog", "DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED": "false",}],
+)
+def test_datadog_128_bit_propagation(test_agent, test_library):
+    """ Ensure that external 128-bit TraceIds are properly propagated in Datadog
+    headers.
+    """
+    with test_library:
+        make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["x-datadog-trace-id", "1234567890123456789"],
+                ["x-datadog-parent-id", "987654321"],
+                ["x-datadog-tags", "_dd.p.tid=640cfd8d00000000"],
+            ],
+        )
+
+    span = get_span(test_agent)
+    trace_id = span.get("trace_id")
+    dd_p_tid = span["meta"].get("_dd.p.tid")
+
+    assert trace_id == 1234567890123456789
+    validate_dd_p_tid(dd_p_tid)
+    assert dd_p_tid == "640cfd8d00000000"
+
+
+@pytest.mark.parametrize(
+    "library_env", [{"DD_TRACE_PROPAGATION_STYLE": "Datadog", "DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED": "false",}],
+)
+def test_datadog_128_bit_generation_disabled(test_agent, test_library):
+    """Ensure that 64-bit TraceIds are properly generated, propagated in
+    datadog headers, and populated in trace data.
+    """
+    with test_library:
+        headers = make_single_request_and_get_inject_headers(test_library, [])
+
+    header_trace_id = headers["x-datadog-trace-id"]
+    span = get_span(test_agent)
+    span_trace_id = span.get("trace_id")
+    dd_p_tid = span["meta"].get("_dd.p.tid")
+
+    # Note: length of 2^64-1 encoded as decimal is 20.
+    assert len(header_trace_id) < 21
+    assert int(header_trace_id, 10) == span_trace_id
```

Comment:
I'm not familiar with Python type handling. Is this conversion enough to ensure that the trace id is less than 2^64-1? What happens, for example, if the value is 2^64+1 (which will still pass the `len(header_trace_id) < 21` test above)?

Reply:
The integers in Python3 are 'unlimited' in size - I've changed the code to make sure that the value is in the correct range rather than just checking length.

Reply:
Thanks!